### PR TITLE
[ISSUE #5595] Add validateSystemTopicWhenUpdateTopic

### LINKED ISSUE #5595: Add brokerConfig which will not validate system topic when update topic
validateSystemTopicWhenUpdateTopic

https://github.com/apache/rocketmq/blob/1cdcce1988e927e4f705f67c8c9683888c0606cf/broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java#L415-L419

### PR DESCRIPTION
**Pull request by drpmma**
Close #5595 

## What is the purpose of the change

Add validateSystemTopicWhenUpdateTopic

### CODE REVIEW

**Review by shy-share — COMMENTED** [review]
I checked the relevant source code and it seems that only the test class calls this method(setTransactionOpBatchInterval), which does not feel very necessary

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java [hunk 1]
```diff
@@ -412,10 +412,12 @@ private synchronized RemotingCommand updateAndCreateTopic(ChannelHandlerContext
             response.setRemark(result.getRemark());
             return response;
         }
-        if (TopicValidator.isSystemTopic(topic)) {
-            response.setCode(ResponseCode.SYSTEM_ERROR);
-            response.setRemark("The topic[" + topic + "] is conflict with system topic.");
-            return response;
+        if (brokerController.getBrokerConfig().isValidateSystemTopicWhenUpdateTopic()) {
+            if (TopicValidator.isSystemTopic(topic)) {
```
Comment (ShadowySpirits) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:416]: Some other places, such as updateAndCreateStaticTopic, also use TopicValidator#isSystemTopic. Should we deal with them in the same way?

**Comment by drpmma** [comment]
> 

This pr is about `validateSystemTopicWhenUpdateTopic `  instead of  `transactionOpBatchInterval`. Maybe you mistake the changes.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java (hunk 1 above)
Comment (drpmma) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:416]: It's reasonable.

**Review by ShadowySpirits — DISMISSED** [review]
LGTM

**Review by lollipopjin — DISMISSED** [review]
LGTM

**Review by aaron-ai — APPROVED** [review]

File: broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java [hunk 2]
```diff
@@ -456,10 +458,12 @@ private synchronized RemotingCommand updateAndCreateStaticTopic(ChannelHandlerCo
...
+                response.setCode(ResponseCode.SYSTEM_ERROR);
```
Comment (aaron-ai) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java:463]: Merge the judgement into one-line could help us to the unnecessary code indent.

**Pull request merged**